content/articles/how-to-set-up-a-python-web-socket-with-aiohttp/index.md

```diff
@@ -0,0 +1,237 @@
+    {"id":"2","title":"Go to the market"},
+    {"id":"3","title":"Prepare dinner"},
+]
+```
+
+The dummy data will be the server's response when called by the client.
+
+Create a handler function that will send the dummy data.
+
+```python
+async def handle(request):
+    return web.json_response(todos)
+```
+
+Notice that the handler is an `async` function returning `JSON` data.
+
+- Initialize the application, set up the routes with their listeners, and add scripts for starting a server as shown below:
+
+```python
+app = web.Application()
+
+app.add_routes([web.get('/', handle),
+        web.get('/todos', handle)])
+
+if __name__ == '__main__':
+    web.run_app(app)
+```
+
+Here we target the `/` route and the `/todos` route, both of which will be handled by the same function. We are also starting our server by running the application instance.
+
+You can start the server by running the following command on your terminal. Before running the command, ensure you are in the project folder that hosts the `server.py` file.
+
+```bash
+./Scripts/python server.py
+```
+
+![Server response](/engineering-education/how-to-set-up-a-python-web-socket-with-aiohttp/server-start-response.png)
+
+The server is up and running, and as you can the server is listening on port `8080`.
+
+### Building a client with AioHTTP
+
+Create a `client.py` file in the project folder. Next, let's configure the client using the AioHTTP.
+
+Import the `aiohttp` and `asyncio` modules.
+
+```python
+import aiohttp
+import asyncio
+```
+
+- `aiohttp` will create a client session that will listen to the server.
+
+- `asyncio` will initialize an event loop which the client will listen to.
+
+Create a method for creating the client session and listening to the server.
+
+```python
+async def main():
+    async with aiohttp.ClientSession() as session:
+        async with session.get('http://localhost:8080/todos') as response:
+
+            print(response.status)
+            print(await response.json())
+```
+
+Here we are creating a client session, awaiting the response from our local server, and printing out the status and `JSON` response from the server.
+
+Initialize an event loop and run the `main` function in the event loop until completion.
+
+```python
+loop = asyncio.get_event_loop()
+loop.run_until_complete(main())
+```
+The event loop will only close when the `main` function has been fully executed.
+
+### Sending data between client and server
+
+To send the data between the client and the server, we need to run the `client.py` file to receive the dummy data we set from the server.
+
+Open a separate tab of your terminal and run the following command inside the project folder.
+
+```bash
+./Scripts/python client.py
+```
+
+The above command will run the `main` function inside the `client.py` file, and your response should be comparable to:
+
+![Sending data response](/engineering-education/how-to-set-up-a-python-web-socket-with-aiohttp/sending-data-response.png)
```

Comment:
You may paste the output rather than taking a screenshot of the output. Do the same for all the images.